src/kernel/disconnected_transactions.h

```diff
@@ -73,6 +80,15 @@ struct DisconnectedBlockTransactions {
             iters_by_txid.emplace((*block_it)->GetHash(), it);
             cachedInnerUsage += RecursiveDynamicUsage(*block_it);
         }
+
+        // Trim the earliest-added entries until we are within memory bounds.
+        while (!queuedTx.empty() && DynamicMemoryUsage() > m_max_mem_usage) {
+            evicted.emplace_back(queuedTx.front());
```

Comment:
This could potentially be done more efficiently by iterating to find the number to evict, then doing a single `splice()` rather than multiple `emplace_back()`. I doubt it's worth the code complexity though.